chapters/VK_EXT_present_timing/PresentTimeInfo.txt

```diff
@@ -0,0 +1,184 @@
+
+include::{generated}/api/structs/VkAbsolutePresentTimeEXT.txt[]
+
+  * pname:presentID is an application-provided identification value.
+    If non-zero, flink:vkGetPastPresentationTimingEXT will return a
+    slink:VkPastPresentationTimingEXT struct with the timing information
+    associated with the present to the associated swapchain for this
+    call to flink:vkQueuePresentKHR, including this value being in the
+    pname:presentID member of the slink:VkPastPresentationTimingEXT
+    struct.
+    In order to be useful to the application, it should: be unique within
+    some period of time that is meaningful to the application.
+  * pname:targetPresentTime, if non-zero, specifies the earliest time the
+    application wants the image to be displayed to the user.
+    pname:targetPresentTime is a time in nanoseconds, according to the
+    time-domain being used.
+    A value of zero specifies that the presentation engine may: display the
+    image at any time.
+  * pname:idealPresentTime provides an indication to the presentation
+    engine of what the application desires its IPD to become.
+    An application can: set pname:idealPresentTime to the same value
+    as pname:targetPresentTime, indicating that the application
+    desires its IPD to remain steady.
+    A value earlier than
+    pname:targetPresentTime indicates that the application desires to
+    shorten its IPD.
+    A value later than pname:targetPresentTime
+    indicates that the application desires to lengthen its IPD.
+    The presentation engine will provide feedback on this value in the
+    data returned by flink:vkGetPastPresentationTimingEXT.
+  * pname:presentSlop is the period of time in nanoseconds before
+    pname:targetPresentTime that the presentation engine may: display
+    the image.
+
+If pname:targetPresentTime is non-zero, the presentation engine should:
+not display the image to the user at a time earlier than
+(pname:targetPresentTime - pname:presentSlop).
+
+Setting pname:targetPresentTime to zero is useful when the application
+desires to provide a non-zero pname:presentID in order to get back timing
+information from flink:vkGetPastPresentationTimingEXT, but does not wish
+to set a pname:targetPresentTime.
+
+Providing a pname:presentSlop enables the application to indicate to the
+presentation engine to display the image within this period if a vertical
+blanking period occurs on FRR displays.
+
+[NOTE]
+.Note
+====
+The pname:presentSlop is used to avoid unintentionally missing a vertical
+blanking period on FRR displays due to rounding errors or drift between
+clocks.
+A suggested value for pname:presentSlop is half the _Refresh Rate_.
+====
+
+include::{generated}/validity/structs/VkAbsolutePresentTimeEXT.txt[]
+
+--
+
+[open,refpage='VkRelativePresentTimeEXT',desc='Specifying the minimum duration an image should be presented',type='structs']
+--
+
+The sname:VkRelativePresentTimeEXT structure is defined as:
```

Comment:
Maybe it would be a good idea to state explicitly the motivation for a separate relative timing target.

Applications submitting a single frame at a time wouldn't have any use for this struct I think, because they could create a suitable absolute target instead (last `actualPresentTime` + `minPresentDuration`).

However, applications queueing multiple frames at once don't know in advance when the presentation engine will choose to display their image. `VkRelativePresentTimeEXT` allows the application to specify a target relative to the previous actual present time, without having to manually adjust the absolute timing (n+1) depending on feedback n.

Reply:
This would be great for stuff like presenting at a stable 30fps on a 60hz display.

Reply:
I think spec language should refrain from making recommendations or suggestions about usage and intent. I know this PR has quite a few of these, which I might clean up, but I'd rather not add more here. The appendix, which could probably be fleshed out a bit, is a better place for these remarks. Alternatively, one of the large existing .NOTE sections may also be appropriate; I'll look into adding a comment to that effect somewhere.